thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java

```diff
@@ -380,6 +380,9 @@ long createDetectionConfig(@NotNull String payload, long startTime, long endTime
     Long id = this.detectionConfigDAO.save(detectionConfig);
     Preconditions.checkNotNull(id, "Error while saving the detection pipeline");
 
+    // create an yaml onboarding task to run replay and tuning
+    createYamlOnboardingTask(detectionConfig.getId(), startTime, endTime);
```

Comment:
Will this cause lot of system burden if user bulk onboarding using API?

Reply:
The onboarding jobs complete within half a minute on an average case and I don't think it should be a problem. In addition, I am not sure if bulk-onboarding of alerts is a common use-case or whether we should even encourage such use-cases except for some special cases like "onboard all UMP metrics" in which case we should anyways take precaution. Alerts are typically created by users on various portals and the throughput(requests/sec) is typically not that high.

Reply:
One approach could be having a flag - like 'ui=true' - in the API, so that the frontend could get the error messages and non-UI clients can skip the messages.

Reply:
> The onboarding jobs complete within half a minute on an average case and I don't think it should be a problem. In addition, I am not sure if bulk-onboarding of alerts is a common use-case or whether we should even encourage such use-cases except for some special cases like "onboard all UMP metrics" in which case we should anyways take precaution. Alerts are typically created by users on various portals and the throughput(requests/sec) is typically not that high.
> 
> On the other hand, if we do not trigger the onboarding job, then apart from the replay, even the tuning component isn't triggered.

Definitely this is not encouraged. We need to have a fuse mechanism somewhere to avoid bringing down the whole site if it happens. Maybe we should throttle the replay tasks.

Reply:
@xiaohui-sun +1, I have filed a separate ticket to track the fuse mechanism.

Reply:
Thanks!